src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/ArmorTask.java

```diff
@@ -119,9 +120,26 @@ private boolean hasSunlight(Player p) {
         return (world.getTime() < 12300 || world.getTime() > 23850) && p.getEyeLocation().getBlock().getLightFromSky() == 15;
     }
 
-    private void checkForRadiation(Player p) {
+    private void checkForRadiation(PlayerProfile profile) {
+        HashedArmorpiece[] armor = profile.getArmor();
+        Player p = profile.getPlayer();
+
         // Check for a Hazmat Suit
-        if (!SlimefunUtils.isItemSimilar(p.getInventory().getHelmet(), SlimefunItems.SCUBA_HELMET, true) || !SlimefunUtils.isItemSimilar(p.getInventory().getChestplate(), SlimefunItems.HAZMAT_CHESTPLATE, true) || !SlimefunUtils.isItemSimilar(p.getInventory().getLeggings(), SlimefunItems.HAZMAT_LEGGINGS, true) || !SlimefunUtils.isItemSimilar(p.getInventory().getBoots(), SlimefunItems.RUBBER_BOOTS, true)) {
+        boolean hasHazmat = false;
+        for (HashedArmorpiece armorPiece : armor) {
+
+            Optional<SlimefunArmorPiece> sfArmor = armorPiece.getItem();
+            if (!sfArmor.isPresent()) continue;
+
+            if (sfArmor.get().getID().equals("SCUBA_HELMET") ||
+                    sfArmor.get().getID().equals("HAZMAT_CHESTPLATE") ||
+                    sfArmor.get().getID().equals("HAZMAT_LEGGINGS") ||
+                    sfArmor.get().getID().equals("RUBBER_BOOTS")) {
+                hasHazmat = true;
```

Comment:
Also since you are only checking for ANY hazmat piece, you can break the loop here.
Or even better, make a seperate method with a boolean return type for this.